presto-tpcds/src/main/java/com/facebook/presto/tpcds/TpcdsRecordSet.java

```diff
@@ -57,7 +57,10 @@ public TpcdsRecordSet(Results results, List<Column> columns)
         this.columns = ImmutableList.copyOf(columns);
         ImmutableList.Builder<Type> columnTypes = ImmutableList.builder();
         for (Column column : columns) {
-            columnTypes.add(getPrestoType(column.getType()));
+            // The TPCDS config tpcds.use-varchar-type is set to true only for native execution, since char type is
+            // currently unsupported in Presto native. This function is only used by the Presto java TPCDS connector
+            // so the argument useVarcharType is false here.
+            columnTypes.add(getPrestoType(column.getType(), false));
```

Comment:
This should use the tpcds.use-varchar-type config. Java should support both options.